Runtime/Core/Smoothing.cs

```diff
@@ -21,24 +21,14 @@ public static class Smoothing
         internal const int smoothingGroupNone = 0;
 
         /// <summary>
-        /// Smoothing groups 1-24 are smooth.
+        /// Smoothing groups 1-30 are smooth.
         /// </summary>
         internal const int smoothRangeMin = 1;
 
         /// <summary>
-        /// Smoothing groups 1-24 are smooth.
-        /// </summary>
-        internal const int smoothRangeMax = 24;
-
-        /// <summary>
-        /// Smoothing groups 25-42 are hard. Note that this is obsolete, and generally hard faces should be marked smoothingGroupNone.
+        /// Smoothing groups 1-30 are smooth.
         /// </summary>
-        internal const int hardRangeMin = 25;
-
-        /// <summary>
-        /// Smoothing groups 25-42 are hard. Note that this is soon to be obsolete, and generally hard faces should be marked smoothingGroupNone.
-        /// </summary>
-        internal const int hardRangeMax = 42;
+        internal const int smoothRangeMax = 30;
```

Comment:
The concept of "smoothing groups" as it is implemented now is a little ridiculous. The concept was borrowed from old school modelers that used a bitset to define smoothing groups, allowing for more complicated merging of adjacent face normals. Our implementation doesn't handle that, so we end up with arbitrary numbers that only exist to define groups. There's no reason to limit this to 30 or any other number.

I guess for the sake of this PR 30 is better than 24, so :woman_shrugging: . Really though we should just remove this whole interface and let users set edges as hard or soft.

Reply:
Yeap I agree, do we have time though to make that change here ?

Reply:
I think it's out of scope for a bug fix PR